### PR TITLE
Add Change Request and Update Log

### LINKED ISSUE #207: Build My Changes module for Dashboard
Description : In our developer dashboards we want to see a third module (or maybe modify My Production) that represents applications that are we are responsible for and what states of Change Management are they in.

Depends ON:#205

Done looks like:

- [ ] Module on Dashaboard
- [ ] Green/red light for success and failure
- [ ] Accessibility Scan, Security Scan, Legal Counsel, Back OUt Plan Completed, Testing Plans Completed, Approved by Change Management.

### LINKED ISSUE #272: Create new model called ChangeRequest

### LINKED ISSUE #274: Connect Upgrade, Change, to Software Record

### PR DESCRIPTION
**Pull request by scherztc**
Fixes #272 and Fixes #274 and Fixes #207

1. The purpose of this PR is to introduce the idea of a ChangeRequest.  
2. The ChangeRequest is assigned to a SoftwareRecord.
3. The ChangeRequest has the following fields () and can be open or closed.
4. Creating a ChangeRequest can be done from the Sidebar Menu on the left hand side.  
5. You can also Create a ChangeRequest from the SoftwareRecord Chang Log tab.
6. You can view/edit all ChangeRequest from the sidebar on the left hand side.
7. You can view/edit the OPEN ChangeRequest associated with a software in the Change Log tab of the Software Record.
8. You can view/edit the CLOSED ChangeRequest associated with a software in the Upgrade History tab of the Software Record.
9. You can view/edit YOUR ChangeRequest in the Dashboard Module.
10. You must be a Manager / Administrator / Technician to CRUD ChangeRequest.  

Create new issues for:

- Improve labelling - Done with PR Review.
- Improve Output format/style - Done with PR Review
- Improve by adding more fields.
- Improve Test - Done with PR Review

### CODE REVIEW

**Review by crowesn — CHANGES_REQUESTED** [review]
This is for UI issues found while testing the feature manually. Code review pending.
 
1. ChangeRequest#edit broken locale?
Screen Shot 2023-08-11 at 12 36 16 PM

2. Change request new/edit form extends above top nav hiding fields. Layout seems to be for mobile but it's not responsive/doesn't change with viewport resizing
Screen Shot 2023-08-11 at 12 38 04 PM

3. Seems like all the Change Request views have this mobile static sizing
Screen Shot 2023-08-11 at 12 40 09 PM

4. On the Software Record show view, consider grouping the change request table and new change request buttons.
Screen Shot 2023-08-11 at 12 42 28 PM

5. The select for change request software record id is missing a label.
Screen Shot 2023-08-11 at 12 52 11 PM

**Review by crowesn — CHANGES_REQUESTED** [review]
Change_request access/auth by user role needs to be expanded and refined using patterns outlined in petergate [documentation](https://github.com/elorest/petergate). There are some conditionals in one of the views that hint at what levels of access each role should have. Add role contexts/shared examples to specs to further document access/auth for each role.

**Comment by scherztc** [comment]
https://github.com/uclibs/application_portfolio/wiki/Roles-in-Application-Portfolio

File: app/views/software_records/_upgrade_history.html.erb [hunk 1]
```diff
@@ -1 +1,49 @@
...
+            <tr>
+              <td class="text-left" style="padding-left: 50px"><%= change_request.change_submitted_date %></td>
+              <td class="text-left" style="padding-left: 50px"><%= change_request.change_title %></td>
+              <td class="text-left" style="padding-left: 50px"><%= find_software_version(change_request.software_record_id) %></td>
+              <td class="text-left" width="250px" style="padding-left: 50px"><% find_tech_leads(change_request.software_record_id).each do |a| %><li><%= a %></li><% end %></td>
+              <% if current_user.role.to_s == "viewer" %>
+                <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
+              <% elsif current_user.role.to_s == "owner" %>
+                <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
+                <td><%= link_to 'Edit', edit_change_request_path(change_requests), { :class => "btn btn-primary" } %></td>
```
Comment (crowesn) [inline app/views/software_records/_upgrade_history.html.erb:35]:
```
ActionView::Template::Error (undefined local variable or method `change_requests' for #<ActionView::Base:0x0000000000e9e8>
Did you mean?  change_request
               change_requests_url
               change_request_url):
    32:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    33:               <% elsif current_user.role.to_s == "owner" %>
    34:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    35:                 <td><%= link_to 'Edit', edit_change_request_path(change_requests), { :class => "btn btn-primary" } %></td>
    36:               <% else %>
    37:                 <td><%= link_to 'View', change_request , { :class => "btn btn-success" }%></td>
    38:                 <td><%= link_to 'Edit', edit_change_request_path(change_request), { :class => "btn btn-primary" } %></td>
  
app/views/software_records/_upgrade_history.html.erb:35
app/views/software_records/_upgrade_history.html.erb:25
app/views/software_records/show.html.erb:59
```

Comment (crowesn) [reply app/views/software_records/_upgrade_history.html.erb:35]: little typo: argument "change_requests" in `edit_change_request_path` helper should be singular "change_request"

**Review by crowesn — CHANGES_REQUESTED** [review]
This is ready to go aside from one small runtime error bug.

**Review by crowesn — APPROVED** [review]

**Pull request merged**